vmdb/app/controllers/provider_foreman_controller.rb

```diff
@@ -71,14 +71,23 @@ def refresh
   def provision
     assert_privileges("provider_foreman_provision")
     @prov_id = find_checked_items
-    @prov_id = params[:id] if @prov_id.empty?
-
-    render :update do |page|
-      page.redirect_to :controller     => "miq_request",
-                       :action         => "prov_edit",
-                       :prov_id        => @prov_id,
-                       :org_controller => "configured_system",
-                       :escape         => false
+    @prov_id.push(params[:id]) if @prov_id.empty?
+
+    provision_configured_systems = true
+    provision_configured_systems =
+      ConfiguredSystem.common_configuration_profiles_for_selected_configured_systems(@prov_id) if @prov_id.size > 1
```

Comment:
I think @brandondunne is suggesting that you let the model make the decision about available configuration profile since a bad setup within Foreman could lead to no profiles for a single system.  For example the organization for a system may not have any profiles defined..

If you remove that check then the need for the `provision_configured_systems` variable goes away and the check would become:
`if ConfiguredSystem.common_configuration_profiles_for_selected_configured_systems(@prov_id)`

You would also want to update the flash message to use "system(s)".  Not sure if you have a better way you normally deal with this.